### PR TITLE
EP: Add `LibSignature` library

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

Adds the `LibSignature` library for signature validation that will be inlined by mtxs and limit orders.

There's also a little bit of relevant tooling in `zero-ex/src/signature_utils` and `zero-ex/src/revert_errors`. I'm suggesting putting this stuff in the contract package for now until we can sort out protocol-only utils and types packages that belong to this repo.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "scripts": {
         "build": "yarn pre_build && tsc -b",
         "build:ci": "yarn build",
+        "build:ts": "tsc -b",
```
Comment (smarx) [inline contracts/zero-ex/package.json:15]: Nitpick: maybe rewrite "build" to use "build:ts"?

**Review by smarx — APPROVED** [review]
LGTM

**Review by hysz — APPROVED** [review]
LGTM, nice work!

File: contracts/zero-ex/package.json [hunk 2]
```diff
@@ -10,8 +10,9 @@
         "test": "test"
     },
     "scripts": {
-        "build": "yarn pre_build && tsc -b",
+        "build": "yarn pre_build && yarn build",
```
[the remark this answers is not in the document]
Comment (dorothy-zbornak) [reply contracts/zero-ex/package.json]: lol yeah just saw that. was trying to figure out why CI was taking forever

**Pull request merged**